### PR TITLE
Merge code from Trust

### PR DESCRIPTION
**Pull request by hewigovens**
* Fix eth_sign and personal_sign
* Support eth_signTransaction and eth_sendTransaction
* Expose error handler

### CODE REVIEW

File: WalletConnect/WCInteractor.swift [hunk 1]
```diff
@@ -249,15 +260,19 @@ extension WCInteractor {
...
+                if let event = WCEvent(rawValue: method) {
+                    try handleEvent(event, topic: topic, decrypted: decrypted)
+                } else if let id = json["id"] as? Int64 {
+                    onCustomRequest?(id, json)
+                }
             }
         } catch let error {
-            print(error)
+            onError?(error)
+            print("==> onReceiveMessage error: \(error.localizedDescription)")
```
Comment (rockfridrich) [inline WalletConnect/WCInteractor.swift:275]: Let's remove all prints from the production code

File: WalletConnect/WCInteractor.swift [hunk 2]
```diff
@@ -249,15 +260,19 @@ extension WCInteractor {
             let decrypted = try WCEncryptor.decrypt(payload: payload, with: session.key)
             guard let json = try JSONSerialization.jsonObject(with: decrypted, options: [])
                 as? [String: Any] else {
-                throw WCError.badServerResponse
+                throw WCError.badJSONRPCRequest
             }
             print("<== decrypted: \(String(data: decrypted, encoding: .utf8)!)")
```
Comment (rockfridrich) [inline WalletConnect/WCInteractor.swift:265]: We should not add prints to the repo

File: WalletConnect/WCInteractor.swift [hunk 3]
```diff
@@ -161,58 +165,65 @@ extension WCInteractor {
...
-        } catch let error {
-            print("==> handleEvent error: \(error.localizedDescription)")
+            break
+        case .bnbTransactionConfirm:
+            let request: JSONRPCRequest<[WCBinanceTxConfirmParam]> = try event.decode(decrypted)
+            guard request.params.count > 0 else {
+                throw WCError.badJSONRPCRequest
+            }
+            bnbTxConfirmResolvers[request.id]?.fulfill(request.params[0])
+            bnbTxConfirmResolvers[request.id] = nil
```
Comment (rockfridrich) [inline WalletConnect/WCInteractor.swift:216]: Let's move all Binance Dex and Ethereum related logic to separate classes.
`WCInteractor` is now responsible for `WalletConnect`, `Ethereum`, and `Binance` logic. We should decouple it to three separate classes/objects. 
This state is definitely should not be related to WCInteractor — https://github.com/WalletConnect/swift-walletconnect-lib/blob/9e1d21ba267f398b5bf4019c76884535cd869628/WalletConnect/WCInteractor.swift#L49

File: WalletConnect/WCInteractor.swift (hunk 1 above)
Comment (rockfridrich) [inline WalletConnect/WCInteractor.swift:275]: Let's remove prints from library code.

File: WalletConnect/WCInteractor.swift [hunk 4]
```diff
@@ -161,58 +165,65 @@ extension WCInteractor {
...
+                onBnbSign?(request.id, request.params[0])
+            } else if let request: JSONRPCRequest<[WCBinanceTransferOrder]> = try? event.decode(decrypted) {
+                onBnbSign?(request.id, request.params[0])
             }
-        } catch let error {
-            print("==> handleEvent error: \(error.localizedDescription)")
+            break
+        case .bnbTransactionConfirm:
+            let request: JSONRPCRequest<[WCBinanceTxConfirmParam]> = try event.decode(decrypted)
+            guard request.params.count > 0 else {
```
Comment (rockfridrich) [inline WalletConnect/WCInteractor.swift:212]: What is the difference of validating either `count` or `first` object property? Could we make validation more generalized and move it to `decode` function?

**Review by rockfridrich — CHANGES_REQUESTED** [review]

File: WalletConnect/WCInteractor.swift (hunk 1 above)
Comment (hewigovens) [reply WalletConnect/WCInteractor.swift:275]: These logs should be only available when debugging: https://github.com/WalletConnect/swift-walletconnect-lib/blob/master/WalletConnect/WCInteractor.swift#L20

File: WalletConnect/WCInteractor.swift (hunk 3 above)
Comment (hewigovens) [reply WalletConnect/WCInteractor.swift:216]: what's the best way to do? add another `EthereumInteractor` and `BinanceInteractor`?

File: WalletConnect/WCInteractor.swift (hunk 4 above)
Comment (hewigovens) [reply WalletConnect/WCInteractor.swift:212]: it depends on each event, some require two parameters, and some require only one. `decode` is generic method and doesn't know about types

**Comment by hewigovens** [comment]
close for this now, will prepare new pr

**Pull request closed without merging**